micromanager_gui/main_window.py

```diff
@@ -271,11 +271,48 @@ def _update_mda_engine(self, newEngine: PMDAEngine, oldEngine: PMDAEngine):
         newEngine.events.sequenceStarted.connect(self._on_mda_started)
         newEngine.events.sequenceFinished.connect(self._on_mda_finished)
 
+    @ensure_main_thread
```

Comment:
This is not necessary on the demo camera on linux. but is on a real scope on windows. :shrug: